core/src/OAuth2/OAuth2Client.cs

```diff
@@ -105,6 +105,28 @@ internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, R
 
                 httpEvent.HttpResponseStatus = (int)response.StatusCode;
                 httpEvent.UserAgent = response.UserAgent;
+                
+                httpEvent.HttpMethod = method.Method.ToString();
+
+                if(response.HeadersAsDictionary.ContainsKey("x-ms-request-id") &&
```

Comment:
HeadersAsDictioary copies the headers into a dictionary. Not an very expensive operation, but since it's on the critical path, could you store this dictionary as a local method variable and reuse it?

Reply:
done